clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerUtils.java

```diff
@@ -231,6 +231,8 @@ public static <T> T getResult(Future<T> future) {
         try {
             return future.get();
         } catch (ExecutionException e) {
+            if (e.getCause() instanceof IllegalStateException)
```

Comment:
interesting, good catch by the integration test. The trick is that this would of course apply to all calls, not only to the assign, but feels like the right thing to ensure we propagate an IllegalState as such, and not obfuscated in a KafkaException.

Reply:
Do we need to have the same logic here as in the `getResult()` on line 218?

Reply:
I updated it as well. Let's wait for CI result. Thanks for catching it.